exec/java-exec/src/main/codegen/templates/NullReader.java

```diff
@@ -56,11 +56,17 @@ public void copyAsValue(MapWriter writer) {}
 
   public void copyAsValue(ListWriter writer) {}
 
-  <#list vv.types as type><#list type.minor as minor><#assign name = minor.class?cap_first /> 
+  public void copyAsValue(UnionWriter writer) {}
+
+  <#list vv.types as type><#list type.minor as minor><#assign name = minor.class?cap_first />
+  public void read(${name}Holder holder){
+    throw new UnsupportedOperationException("NullReader cannot read into non-nullable holder");
```

Comment:
UserException? I'm not sure what this message even means or what would cause this. I'm thinking it is a dev only message and thus Unsupported would be fine but want to confirm.